src/lib.rs

```diff
@@ -91,3 +91,22 @@ pub trait Voxel {
     /// Returns `true` iff light cannot pass through this voxel.
     fn is_opaque(&self) -> bool;
 }
+
+/// Used as a dummy for functions that must wrap a voxel
+/// but don't want to change the original's properties.
+struct IdentityVoxel<'a, T: Voxel>(&'a T);
+
+impl<'a, T: Voxel> Voxel for IdentityVoxel<'a, T> {
+    fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+    fn is_opaque(&self) -> bool {
+        self.0.is_opaque()
+    }
```

Comment:
Maybe add `#[inline]` hints?

Reply:
I don't know much about inlining, so I'll leave that decision to someone else.

Reply:
General guidance is here: https://nnethercote.github.io/perf-book/inlining.html

My policy has been to `#[inline]` very short functions that might get called in a tight loop. It's just a suggestion to the compiler, and the compiler might decide it's a bad idea to inline a particular function, and I'm OK trusting the compiler's decision in most cases.

Reply:
So I think inlining these trait methods makes sense. Pushing and popping a stack frame in either of these functions would be bad for performance.

Reply:
Done.